databricks/koalas/config.py

```diff
@@ -28,31 +28,42 @@
 __all__ = ['get_option', 'set_option', 'reset_option']
 
 
-# dict to store registered options and their default values (key -> default).
+def _make_check_func(func, msg):
+    def check(v):
+        if func(v):
+            return v
```

Comment:
Hmm, but this way won't allow custom confusion... Let me think a bit more about it